lib/icinga/pluginutility.cpp

```diff
@@ -192,10 +192,25 @@ String PluginUtility::FormatPerfdata(const Array::Ptr& perfdata)
 		else
 			first = false;
 
-		if (pdv.IsObjectType<PerfdataValue>())
+		if (pdv.IsObjectType<PerfdataValue>()) {
 			result << static_cast<PerfdataValue::Ptr>(pdv)->Format();
-		else
+		} else if (normalize) {
+			PerfdataValue::Ptr normalized;
+
+			try {
+				normalized = PerfdataValue::Parse(pdv);
+			} catch (const std::invalid_argument& ex) {
+				Log(LogDebug, "PerfdataValue") << ex.what();
+			}
+
+			if (normalized) {
+				result << normalized->Format();
+			} else {
+				result << pdv;
+			}
```

Comment:
Can't this be just done within the try/catch? Shouldn't make a difference unless `normalized->Format()` is supposed to throw `std::invalid_argument` and this should be passed to the caller.

Reply:
I, however, prefer as less as possible in a try/catch.